### ISSUE TITLE
A11y: Value list item with non-interactive prop should not read in screen reader or be in the tab order

### ISSUE THREAD

**Issue by paulcpederson** [Esri/calcite-components#3209]
### Summary

The `non-interactive` prop should make the value list item completely non-interactive. While currently it prevents clicking via mouse, it's still interactive to keyboard users and is read as an interactive menu item to users using a screen reader.

### Actual Behavior

1. NVDA reads "Clickable _[label]_, _[label]_ menu item, not checked"
2. Label is still in the tab order and has a focus state 

### Expected Behavior

Nothing should be read by the screen reader and the item should not be a menu item. When non-interactive all interactive roles should be removed and element should be removed from the tab order. 

### Reproduction Steps and Sample

_Sample_: https://codepen.io/paulcp/pen/JjyoymE

1. Tab to the value list, note the focus outline
2. Turn on a screen reader, note when tabbed to the label it still has an interactive set of roles and is read aloud as if it's interactive

### Relevant Info

_Version_: `@esri/calcite-components@beta.66`

**Comment by eriklharper** [comment]
What is the difference between `non-interactive` and `disabled`?

**Comment by geospatialem** [comment]
Blocked due to re-work on the `list` component. 

Will revisit, and determine feasibility, or if the `non-interactive` prop will proceed with the updated component structure.

**Comment by geospatialem** [comment]
> Blocked due to re-work on the `list` component.
> 
> Will revisit, and determine feasibility, or if the `non-interactive` prop will proceed with the updated component structure.

@driskull If memory serves, were we thinking of dropping the `nonInteractive` prop from list-based components? Thinking we could either close this out, or revisit post 1.0 release depending on next steps.

**Comment by driskull** [comment]
Yes, its already been removed

**Comment by geospatialem** [comment]
Closing as this is no longer applicable with the removal of the `nonInteractive` prop.
